src/Neo/Network/P2P/Peer.cs

```diff
@@ -365,14 +365,51 @@ private void OnTerminated(IActorRef actorRef)
 
         private void OnTimer()
         {
-            // Check if the number of desired connections is already enough
-            if (ConnectedPeers.Count >= Config.MinDesiredConnections) return;
+            var deficit = Config.MinDesiredConnections - ConnectedPeers.Count;
+
+            if (deficit <= 0 && ConnectingPeers.Count == 0)
+                return;
+
+            if (deficit > 0 && UnconnectedPeers.Count == 0)
+            {
+                NeedMorePeers(deficit);
+                if (UnconnectedPeers.Count == 0)
+                    return;
+            }
+
+            // Check min values
 
-            // If there aren't available UnconnectedPeers, it triggers an abstract implementation of NeedMorePeers
             if (UnconnectedPeers.Count == 0)
-                NeedMorePeers(Config.MinDesiredConnections - ConnectedPeers.Count);
+                return;
+
+            var maxConnections = Config.MaxConnections < 0
+                ? int.MaxValue
+                : Config.MaxConnections - ConnectedPeers.Count;
+            if (maxConnections <= 0)
+                return;
+
+            var connectingCapacity = ConnectingMax - ConnectingPeers.Count;
+            if (connectingCapacity <= 0)
+                return;
+
+            var toConnect = Math.Max(deficit, 0);
```

Comment:
if deficit is `0` and you reached here it is because ConnectingPeers.Count > 0

So, perhaps this needs to be considered